src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -118,13 +122,13 @@ bool InterpreterDropQuery::deleteTableDistributed(const ASTDropQuery & query)
         auto query_str = queryToString(query);
         LOG_INFO(log, "Drop DistributedMergeTree query={} query_id={}", query_str, ctx->getCurrentQueryId());
 
-        std::vector<std::pair<String, String>> string_cols = {
-            {"payload", ctx->getQueryParameters().at("_payload")},
-            {"database", query.database},
-            {"table", query.table},
-            {"query_id", ctx->getCurrentQueryId()},
-            {"user", ctx->getUserName()}
-        };
+        std::vector<std::pair<String, String>> string_cols
+            = {{"payload", ctx->getQueryParameters().at("_payload")},
+               {"database", query.database},
+               {"table", query.table},
+               {"query_id", ctx->getCurrentQueryId()},
+               {"user", ctx->getUserName()},
+               {"password", ctx->getUser()->authentication.getPassword()}};
```

Comment:
We shall not pass `password` in the record header, `password` in general shall not stored in kafka topic. We can just keep the existing schema as it is and get the password in `DDLService` for users by using `ctx->getUser()->authentication.getPassword()` ?

Reply:
fixed